test/verify/check-embed

```diff
@@ -72,6 +74,37 @@ Shell=/shell/index.html
         b.switch_to_frame("embed-auth")
         b.wait_visible("#login-user-input")
 
+    @skipBrowser("Chromium cannot inspect cross-origin frames", "chromium")
+    def testCrossOrigin(self):
+        b = self.browser
+        m = self.machine
+
+        pyhttpd = subprocess.Popen(['python3', '-m', 'http.server', '--bind=localhost',
+                                    '--directory=test/verify/files/embed-cockpit', '12346'])
+        self.addCleanup(pyhttpd.wait)
+        self.addCleanup(pyhttpd.terminate)
+
+        # log in normally, to get the auth cookie into the browser and thus maximize possible cross-domain exposure
+        self.login_and_go()
+
+        b.open("http://localhost:12346/index.html")
+        b.set_val("#embed-address", "http://{0}:{1}".format(m.web_address, m.web_port))
+        b.click("#embed-full")
+        # FIXME (#16122): we should not even get that far, frame loading should already be blocked here
+        b.wait_visible("iframe[name='embed-full'][loaded]")
+        b.switch_to_frame("embed-full")
+
+        # second line of defense: existing login cookie does not work (default browser protection)
+        b.wait_visible("#login")
+        b.set_val("#login-user-input", "admin")
+        b.set_val("#login-password-input", "foobar")
+        b.click('#login-button')
+        b.expect_load_frame("embed-full")
+
+        # third line of defense: login succeeds and creates a PAM session; but loading session UI does not
+        # (again, due to default browser protection)
+        b.wait_visible("#login")
```

Comment:
Is there a race here? ('#login' might still be visible here and the real page might load later if the login actually worked.)
Can we wait for #login-user-input to be empty again, for example?

Reply:
It's a race condition either way, I'm afraid -- if we wait for `#login` to *not* be present, then the failure and re-draw of login might happen faster than the browser and our JS/Python polling machinery to pick it up. My hope here was that waiting for the frame to load in line 102) would already be correct, and that after that there can't be any "old" contents.

Note that the next commit removes all that code anyway.

Reply:
Yeah, waiting for something to disappear and then reappear doesn't work.  Does expect_load_frame do anything when the frame is already loaded?

One idea would be to add something to the DOM artificially in the test before clicking #login, and then wait for it to disappear (which only happens when the page is reloaded.)

> Note that the next commit removes all that code anyway.

Heh, true.

Reply:
> Does expect_load_frame do anything when the frame is already loaded?

yes, it literally waits for a "frame loaded" event. If there is none, it will hang and time out, and fail the test. It's edge triggered (unlike our `wait_present()` and friends, which is level-triggered)

Reply:
I ran this test just once in CI (successfully), and lots of times locally. This was mostly to play around with the original behaviour, what the CVE complained about.

Reply:
I'm not changing this -- I am reasonably convinced that the explicit `expect_load_frame()` DTRT here.

Reply:
Agreed.